Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/ConvertToWavelength.py

```diff
@@ -0,0 +1,146 @@
+
+        progress = Progress(self, start=0.0, end=1.0, nreports=3)
+
+        # Convert the units into wavelength
+        progress.report("Converting workspace to wavelength units.")
+        workspace = self._convert_units_to_wavelength(workspace)
+
+        # Get the rebin option
+        rebin_type = RebinType.from_string(self.getProperty("RebinMode").value)
+        rebin_string = self._get_rebin_string(workspace)
+        if rebin_type is RebinType.Rebin:
+            rebin_options = {"InputWorkspace": workspace,
+                             "PreserveEvents": True,
+                             "Params": rebin_string}
+        else:
+            rebin_options = {"InputWorkspace": workspace,
+                             "Params": rebin_string}
+
+        # Perform the rebin
+        progress.report("Performing rebin.")
+        workspace = self._perform_rebin(rebin_type, rebin_options, workspace)
+
+        append_to_sans_file_tag(workspace, "_toWavelength")
+        self.setProperty("OutputWorkspace", workspace)
+        progress.report("Finished converting to wavelength.")
+
+    def validateInputs(self):
+        errors = dict()
+        # Check the wavelength
+        wavelength_low = self.getProperty("WavelengthLow").value
+        wavelength_high = self.getProperty("WavelengthHigh").value
+        if wavelength_low is not None and wavelength_high is not None and wavelength_low > wavelength_high:
+            errors.update({"WavelengthLow": "The lower wavelength setting needs to be smaller "
+                                            "than the higher wavelength setting."})
+
+        if wavelength_low is not None and wavelength_low < 0:
+            errors.update({"WavelengthLow": "The wavelength cannot be smaller than 0."})
+
+        if wavelength_high is not None and wavelength_high < 0:
+            errors.update({"WavelengthHigh": "The wavelength cannot be smaller than 0."})
+
+        wavelength_step = self.getProperty("WavelengthStep").value
+        if wavelength_step is not None and wavelength_step < 0:
+            errors.update({"WavelengthStep": "The wavelength step cannot be smaller than 0."})
+
+        # Check the workspace
+        workspace = self.getProperty("InputWorkspace").value
+        rebin_type = RebinType.from_string(self.getProperty("RebinMode").value)
+        if rebin_type is RebinType.InterpolatingRebin and isinstance(workspace, EventWorkspace):
+            errors.update({"RebinMode": "An interpolating rebin cannot be applied to an EventWorkspace."})
+        return errors
+
+    def _convert_units_to_wavelength(self, workspace):
+        convert_name = "ConvertUnits"
+        convert_options = {"InputWorkspace": workspace,
+                           "Target": "Wavelength"}
+        convert_alg = create_unmanaged_algorithm(convert_name, **convert_options)
+        convert_alg.setPropertyValue("OutputWorkspace", EMPTY_NAME)
+        convert_alg.setProperty("OutputWorkspace", workspace)
```

Comment:
The algorithms are run in general with unnamed workspaces, ie workspaces which are not placed on the ADS. You might have seen that the `OutputWorkspace` property needs to be set, something like
```python
...
alg.setProperty("InputWorkspace", ws)
...
alg.setProperty("OutputWorkspace", "test")
alg.execute()
out_ws = alg.getProperty("OutputWorkspace").value
....
```
Mantid needs names for the `OutputWorkspace` property even if they are only temporary. It also means that output workspace is most likely a copy of the input. In my case here I want to perform an in-place operation on an unnamed workspace, hence I need to provide the workspace itself to the output and I need to provide a temporary name.

The example below might illustrate this
```python
# -----------------------------------------------------------------------------------------------------------------------
# Create a fake algorithm which peforms some in-place operation
# -----------------------------------------------------------------------------------------------------------------------
class FakeAlgorithm(DataProcessorAlgorithm):
    def PyInit(self):
        self.declareProperty(MatrixWorkspaceProperty("InputWorkspace", '', optional=PropertyMode.Mandatory, direction=Direction.Input), doc='input')
    def PyExec(self):
        ws = self.getProperty("InputWorkspace").value
        
        alg = AlgorithmManager.createUnmanaged("ConvertUnits")
        alg.initialize()
        alg.setChild(True)
        alg.setProperty("InputWorkspace", ws)
        alg.setProperty("Target", "Wavelength")
        # UNCOMMENT LINE BELOW TO MAKE IT WORK
        #alg.setPropertyValue("OutputWorkspace", "EMPTY")
        alg.setProperty("OutputWorkspace", ws)
        
        alg.execute()
        ws_converted = alg.getProperty("OutputWorkspace").value

AlgorithmFactory.subscribe(FakeAlgorithm)


# -----------------------------------------------------------------------------------------------------------------------
# Provide an unnamed workspace
# -----------------------------------------------------------------------------------------------------------------------
alg_created = AlgorithmManager.createUnmanaged("CreateSampleWorkspace")
alg_created.initialize()
alg_created.setChild(True)
alg_created.setProperty("OutputWorkspace", "dummy")
alg_created.execute()
out_ws = alg_created.getProperty("OutputWorkspace").value


# -----------------------------------------------------------------------------------------------------------------------
# Run the algorithm
# -----------------------------------------------------------------------------------------------------------------------
alg_fake = AlgorithmManager.createUnmanaged("FakeAlgorithm")
alg_fake.initialize()
alg_fake.setChild(True)
alg_fake.setProperty("InputWorkspace", out_ws)
alg_fake.execute()
```
You should see that if you uncomment the marked line, then the algorithm runs without an issue, else it will complain that there has been no name set for the output